metrics/runtime_metrics_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package metrics
+
+import (
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber-go/tally"
+)
+
+var _emptyRuntimeConfig = RuntimeConfig{}
+
+func TestRuntimeCollector(t *testing.T) {
+	collector := NewRuntimeCollector(tally.NoopScope, time.Millisecond)
+	defer closeCollector(t, collector)
+
+	assert.False(t, collector.IsRunning())
+	collector.Start()
+	assert.True(t, collector.IsRunning())
+	runtime.GC()
+	time.Sleep(5 * time.Millisecond)
+	collector.generate()
```

Comment:
No need for mocks here either.

Just right above in this test, we are using `tally.Noopscope`. Instead of that we should have a `memReporter` that just creates a hash of reported metrics (I think we should already have that somewhere, and if not that's pretty easy to build and we can rebase onto that diff).

Then you don't have to mock anything, you can assert on that reporter directly to make sure it has the correct metrics.